src/main/java/edu/ie3/datamodel/utils/validation/ThermalUnitValidationUtils.java

```diff
@@ -57,6 +60,46 @@ private ThermalUnitValidationUtils() {
     return exceptions;
   }
 
+  /**
+   * Validates a thermal grid if:
+   *
+   * <ul>
+   *   <li>it is not null
+   * </ul>
+   *
+   * A "distribution" method, that forwards the check request to specific implementations to fulfill
+   * the checking task, based on the class of the given object.
+   *
+   * @param thermalGrid ThermalGrid to validate
+   * @return a list of try objects either containing an {@link ValidationException} or an empty
+   *     Success
+   */
+  protected static List<Try<Void, ? extends ValidationException>> check(ThermalGrid thermalGrid) {
+    Try<Void, InvalidEntityException> isNull = checkNonNull(thermalGrid, "a thermal grid");
+
+    if (isNull.isFailure()) {
+      return List.of(isNull);
+    }
+
+    List<Try<Void, ? extends ValidationException>> exceptions = new ArrayList<>();
+
+    // Validate houses
+    for (ThermalHouseInput house : thermalGrid.houses()) {
+      exceptions.addAll(checkThermalHouse(house));
+    }
+
+    // Validate storages
+    for (ThermalStorageInput storage :
+        Stream.concat(
+                thermalGrid.heatStorages().stream(),
+                thermalGrid.domesticHotWaterStorages().stream())
```

Comment:
Both fields are not present in the `ThermalGrid`. The PR #1132 should therefore be merged before.